File: Easy/Chuck Norris/Chuck Norris.cs
using System;
using System.Linq;
using System.Text;

class Solution
{
    static void Main(string[] args)
    {
        string message = Console.ReadLine();

        // Example: "C" >> "67"
        byte[] bArray = Encoding.ASCII.GetBytes(message);

        string binary = String.Empty;
        foreach (byte bit in bArray)
        {
            // The input message consists of ASCII characters (7-bit)!
            binary += Convert.ToString(bit, 2).PadLeft(7, '0');
        }

        int[] intArray = binary.Select(o => Convert.ToInt32(o)).ToArray();

        // It means the beginning of text
        int lastPosition    = 2;
        string result       = "";

        // ASCII >> 49 = 1; 48 = 0
        for (int i = 0; i < intArray.Length; i++)
        {
            if (intArray[i] == 49)
            {
                if (lastPosition == 1)
                {
                    result += "0";
                }
                else if (lastPosition == 0)
                {
                    result += " 0 0";
                }
                else // beginning
                {
                    result += "0 0";
                }
                lastPosition = 1; // new last bit
            }
            else
            {
                if (lastPosition == 0)
                {
                    result += "0";
                }
                else if (lastPosition == 1)
                {
                    result += " 00 0";
                }
                else // beginning
                {
                    result += "00 0";
                }
                lastPosition = 0; // new last bit
            }
        }
        Console.WriteLine(result);
    }
}

File: Easy/Chuck Norris/JavaScript/Chuck Norris.js
message = readline();

// Example: "C" >> "67"
bArray = [];

for (i = 0; i < message.length; ++i)
    bArray[i] = message.charCodeAt(i);

binary = "";
for (i = 0; i < bArray.length; i++) {
    // The input message consists of ASCII characters (7-bit)!
    temp = bArray[i].toString(2);
    temp = "0000000".substring(0, 7 - temp.length) + temp;
    binary += temp;
}

intArray = []
strArray = binary.split("");

for (i = 0; i < strArray.length ; i++)
    intArray[i] = parseInt(strArray[i]);

// It means the beginning of text
lastPosition = 2;
result = "";

for (i = 0; i < intArray.length; i++) {
    if (intArray[i] == 1)
        result += lastPosition == 1 ? "0" : lastPosition == 0 ? " 0 0" : "0 0";
    else
        result += lastPosition == 0 ? "0" : lastPosition == 1 ? " 00 0" : "00 0";

    lastPosition = intArray[i] == 1 ? 1 : 0; // new last bit
}
print(result);